IBM/openapi-validator
# refactor: use Spectral to check for unused definitions

Purpose:
- Simplify validator rulest by using equivalent Spectral rule

Changes:
- Enable the oas2-unused-definition and oas3-unused-components-schema Spectral rules
- Remove the in-code validation for unused definitions

Tests:
- Remove irrelevant tests used to ensure mentioned Spectral rules are disabled
- Remove unused definitions from test files used to ensure the mentioned Spectral rules are disabled
- Remove irrelevant tests for in-code validation for unused definitions
